test/helpers/position_assertions.h

```diff
@@ -92,6 +99,31 @@ class ErrorAssertion final : public RangeAssertion {
     bool check(const Diagnostic &diagnostic, std::string_view sourceLine, std::string_view errorPrefix);
 };
 
+class UntypedAssertion final : public RangeAssertion {
+public:
+    static std::shared_ptr<UntypedAssertion> make(std::string_view filename, std::unique_ptr<Range> &range,
+                                                  int assertionLine, std::string_view assertionContents,
+                                                  std::string_view assertionType);
+
+    UntypedAssertion(std::string_view filename, std::unique_ptr<Range> &range, int assertionLine,
+                     std::string_view message);
+
+    std::string toString() const override;
+    const std::string message;
+
+    // this exists solely to allow us to reuse ErrorAssertion's checkAll
+    // for UntypedAssertion. It should *always* be false.
+    static constexpr bool matchesDuplicateErrors = false;
```

Comment:
There's other ways to do this without adding this field

eg: something like

```cpp
   if constexpr (std::is_same_v<T,  ErrorAssertion>)
      // ... do things with matchesDuplicateErrors
```
or by adding a template parameter to the `checkAll` template
```cpp
template <typename T, typename F>
checkAll(..., F matchesDuplicateErrors)
```
where `matchesDuplicateErrors: T -> bool` but that's a bit messy as well

So went for this "simple" one
```